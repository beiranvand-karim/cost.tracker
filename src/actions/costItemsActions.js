export const ADD_FILTERED_COST_ITEMS = "ADD_FILTERED_COST_ITEMS";
export const ADD_COST_ITEM = "ADD_COST_ITEM";


export const addFilteredCostItemsAction = data => ({
  type: ADD_FILTERED_COST_ITEMS,
  payload: data
});

export const addCostItemAction = data => ({
  type: ADD_COST_ITEM,
  payload: data
});
