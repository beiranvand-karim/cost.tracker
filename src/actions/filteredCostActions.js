export const ADD_COST = 'ADD_COST';
export const ADD_FILTERED_COSTS = 'ADD_FILTERED_COSTS';

export const addCostAction = (data) => {
   return {
      type: ADD_COST,
      payload: data
   }
};

export const addFilteredActions = (data) => {
   return {
      type: ADD_FILTERED_COSTS,
      payload: data
   }
};
