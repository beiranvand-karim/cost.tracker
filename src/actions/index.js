export {
  addCostItemAction,
  addFilteredCostItemsAction
} from "./costItemsActions";
export { addSourceCostItemAction } from "./sourceCostItemsActions";
