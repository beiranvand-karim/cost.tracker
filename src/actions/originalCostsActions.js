export const ADD_ORIGINAL_COST = 'ADD_ORIGINAL_COST';

export const addOriginalCostAction = (data) => {
   return {
      type: ADD_ORIGINAL_COST,
      payload: data
   }
};
