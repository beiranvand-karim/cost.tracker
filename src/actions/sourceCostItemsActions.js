export const ADD_SOURCE_COST_ITEM = "ADD_SOURCE_COST_ITEM";

export const addSourceCostItemAction = data => ({
  type: ADD_SOURCE_COST_ITEM,
  payload: data
});
