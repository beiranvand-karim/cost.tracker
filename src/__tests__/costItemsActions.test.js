/* global expect, describe, it, beforeEach */
import { addCostItemAction } from "../actions/costItemsActions";
import costItems from "../__helpers__/costItems";
import configureStore from "redux-mock-store";

describe("costItemsActions", () => {
  const mockStore = configureStore();
  const store = mockStore();

  beforeEach(() => {
    store.clearActions();
  });

  it("should match the add cost actions snapshot", () => {
    store.dispatch(addCostItemAction(costItems[0]));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("should match add filtered actions snapshot", () => {
    store.dispatch(addCostItemAction(costItems));
    expect(store.getActions()).toMatchSnapshot();
  });
});
