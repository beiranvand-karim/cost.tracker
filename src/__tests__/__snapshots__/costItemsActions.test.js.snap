// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`costItemsActions should match add filtered actions snapshot 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "category": "CLOTHING",
        "comment": "this is a comment",
        "cost": 1000,
      },
      Object {
        "category": "RENT",
        "comment": "this is a comment",
        "cost": 1500,
      },
      Object {
        "category": "FOOD",
        "comment": "this is a comment",
        "cost": 2000,
      },
    ],
    "type": "ADD_COST_ITEM",
  },
]
`;

exports[`costItemsActions should match the add cost actions snapshot 1`] = `
Array [
  Object {
    "payload": Object {
      "category": "CLOTHING",
      "comment": "this is a comment",
      "cost": 1000,
    },
    "type": "ADD_COST_ITEM",
  },
]
`;
